packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -470,12 +470,13 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
                 part.onDidChangeVisibility(() => widget.suppressUpdateViewVisibility = part.isHidden);
 
                 const tryFireOnDidExpandView = () => {
-                    if (!part.collapsed && part.isVisible) {
+                    if (widget.widgets.length === 0) {
```

Comment:
@danarad05 can you explain what is happening here or why the change is necessary? 
The following fires the `onDidExpandViewEmitter` despite the view being collapsed.

Reply:
@vince-fugnitto 

A) The problem is that there are 2 parts for populating data of a certain view:
1.  onload `prepareViewContainer` is invoked which initiates process
https://github.com/eclipse-theia/theia/blob/eb83e0e00d6c87ca1a98e8a807e7d265b3263a01/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L452

2.  When `TreeViewExtImpl` is created it invokes  `$registerTreeDataProvider` which invokes `registerViewDataProvider`
which retrieves relevant data and runs `prepareView` which in turn adds `viewDataWidget` to widgets array (= loads successfully).

https://github.com/eclipse-theia/theia/blob/eb83e0e00d6c87ca1a98e8a807e7d265b3263a01/packages/plugin-ext/src/main/browser/view/tree-views-main.ts#L57-L58 

B) Now - 
- If you load Theia and stay in `explorer` tab - you enter into first `if` - all is well. 
- If on the other hand you start on `explorer` and more to other tab - then you will enter `else` section - which means that it is waiting for `onDidExpandView` in order to complete loading of view with data.

![image](https://user-images.githubusercontent.com/71069170/112490509-18701780-8d88-11eb-952f-5c686cb9ea51.png)

C) So when you start with `explorer` and switch to another tab, `onDidExpandView` manages to be fired already but `registerViewDataProvider` isn't yet listening for it and so data is not populated.

![image](https://user-images.githubusercontent.com/71069170/112491581-1f4b5a00-8d89-11eb-9b11-688301fa5991.png)

D) I wanted to find an indicator if data was populated or not - and saw that `prepareView` is essentially adding viewDataWidget to widget when successful - so I based distinction on that .

So till widget is added the dataWidget - switching between tabs will invoke `onDidChangeVisibility` which will eventually fire `onDidExpandView` event - an upon it's firing - listeners (onCollapsed, onDidChangeVisibility) will be disposed and so - no additional events will be fired.

Hope I explained it clearly.

Reply:
@danarad05 thanks for explaining, but with the updated changes we are now firing `onDidExpandViewEmitter` even if the view-part (ex: `npm`) is not expanded, is that correct?

---

I added logs and I see that the event is fired for the two use-cases which I believe is incorrect:
- the view is collapsed initially on startup:

```
root INFO --- firing 'onDidExpandViewEmitter' for id='npm', state={ collapsed=true, visible=true } ---
```

- the view is completely hidden on startup:

```
root INFO --- firing 'onDidExpandViewEmitter' for id='npm', state={ collapsed=true, visible=false } ---
```

Logically for both use-cases I do not see why firing `onDidExpandViewEmitter` makes sense (one is collapsed, and the other is completely hidden).

Reply:
@vince-fugnitto 

You are right, it can be improved to filter out those scenarios. Fixed.